src/Symfony/Component/Form/Form.php

```diff
@@ -726,7 +726,15 @@ public function isEmpty()
             }
         }
 
-        return FormUtil::isEmpty($this->modelData) ||
+        if (!method_exists($this->config, 'getIsEmptyCallback')) {
+            @trigger_error(sprintf('Not implementing the "%s::getIsEmptyCallback()" method in "%s" is deprecated since Symfony 5.1.', FormConfigInterface::class, \get_class($this->config)), E_USER_DEPRECATED);
+
+            $isEmptyCallback = null;
+        } else {
+            $isEmptyCallback = $this->config->getIsEmptyCallback();
+        }
+
+        return $isEmptyCallback ? $isEmptyCallback($this->modelData) : FormUtil::isEmpty($this->modelData) ||
```

Comment:
I think it would be more readable this way:

```php
if (null !== $isEmptyCallback) {
    return $isEmptyCallback($this->modelData);
}

return FormUtil::isEmpty($this->modelData) ||
    // arrays, countables
    ((\is_array($this->modelData) || $this->modelData instanceof \Countable) && 0 === \count($this->modelData)) ||
    // traversables that are not countable
    ($this->modelData instanceof \Traversable && 0 === iterator_count($this->modelData));
```